### ISSUE TITLE
Standalone almost-empty polar-bookshelf generates excessive ping lag under a VPN

### ISSUE THREAD

**Issue by bvborisov** [burtonator/polar-bookshelf#793]
**Describe the bug**
When I run polar-bookshelf under a VPN the ping increases 10x (from 40 ms to 3000ms) 

**To Reproduce**
1. Start VPN (I use PIA, but try any other)
2. Measure baseline ping to google dns 8.8.8.8
3. Start polar-bookshelf
4. Observe 10x increase in ping

**Expected behavior**
I expected a small increase in ping (like when no VPN is connected), although any increase is also questionable: does polar establish a remote connection to a server ? Is it not "offline-first" ?  

**Screenshots**
Example of ping increasing after polar is started under VPN:
```
# baseline ping with VPN (connecting to a server in my own region)
64 bytes from 8.8.8.8: icmp_seq=3152 ttl=121 time=40.9 ms
64 bytes from 8.8.8.8: icmp_seq=3153 ttl=121 time=118 ms
64 bytes from 8.8.8.8: icmp_seq=3154 ttl=121 time=47.4 ms
64 bytes from 8.8.8.8: icmp_seq=3155 ttl=121 time=45.8 ms
64 bytes from 8.8.8.8: icmp_seq=3156 ttl=121 time=56.4 ms
64 bytes from 8.8.8.8: icmp_seq=3157 ttl=121 time=53.6 ms
64 bytes from 8.8.8.8: icmp_seq=3158 ttl=121 time=61.6 ms
64 bytes from 8.8.8.8: icmp_seq=3159 ttl=121 time=40.5 ms
64 bytes from 8.8.8.8: icmp_seq=3160 ttl=121 time=68.4 ms
64 bytes from 8.8.8.8: icmp_seq=3161 ttl=121 time=285 ms
# polar (and only polar) started at this point. I had my browser closed.
# the ping becomes quite noticeable, and subsequently trying to use a browser is very slow
64 bytes from 8.8.8.8: icmp_seq=3162 ttl=121 time=1346 ms
64 bytes from 8.8.8.8: icmp_seq=3163 ttl=121 time=1375 ms
64 bytes from 8.8.8.8: icmp_seq=3164 ttl=121 time=1450 ms
64 bytes from 8.8.8.8: icmp_seq=3165 ttl=121 time=1459 ms
64 bytes from 8.8.8.8: icmp_seq=3166 ttl=121 time=1609 ms
64 bytes from 8.8.8.8: icmp_seq=3167 ttl=121 time=1781 ms
64 bytes from 8.8.8.8: icmp_seq=3168 ttl=121 time=2090 ms
64 bytes from 8.8.8.8: icmp_seq=3169 ttl=121 time=2472 ms
64 bytes from 8.8.8.8: icmp_seq=3170 ttl=121 time=2663 ms
64 bytes from 8.8.8.8: icmp_seq=3171 ttl=121 time=2809 ms
64 bytes from 8.8.8.8: icmp_seq=3172 ttl=121 time=2818 ms
64 bytes from 8.8.8.8: icmp_seq=3173 ttl=121 time=2516 ms
64 bytes from 8.8.8.8: icmp_seq=3174 ttl=121 time=2361 ms
```

Example of ping increase after polar is started without vpn is a lot smaller in absolute terms.
```
# baseline ping, no polar, no VPN
64 bytes from 8.8.8.8: icmp_seq=3441 ttl=121 time=11.2 ms
64 bytes from 8.8.8.8: icmp_seq=3442 ttl=121 time=11.4 ms
64 bytes from 8.8.8.8: icmp_seq=3443 ttl=121 time=11.5 ms
64 bytes from 8.8.8.8: icmp_seq=3444 ttl=121 time=11.9 ms
64 bytes from 8.8.8.8: icmp_seq=3445 ttl=121 time=12.4 ms
64 bytes from 8.8.8.8: icmp_seq=3446 ttl=121 time=11.4 ms
64 bytes from 8.8.8.8: icmp_seq=3447 ttl=121 time=11.9 ms
64 bytes from 8.8.8.8: icmp_seq=3448 ttl=121 time=11.3 ms
64 bytes from 8.8.8.8: icmp_seq=3449 ttl=121 time=11.3 ms
64 bytes from 8.8.8.8: icmp_seq=3450 ttl=121 time=11.3 ms
64 bytes from 8.8.8.8: icmp_seq=3451 ttl=121 time=10.6 ms
64 bytes from 8.8.8.8: icmp_seq=3452 ttl=121 time=11.3 ms
64 bytes from 8.8.8.8: icmp_seq=3453 ttl=121 time=11.2 ms
# polar was started around this time. The ping did increase, but not substantially
64 bytes from 8.8.8.8: icmp_seq=3454 ttl=121 time=44.5 ms
64 bytes from 8.8.8.8: icmp_seq=3455 ttl=121 time=13.5 ms
64 bytes from 8.8.8.8: icmp_seq=3456 ttl=121 time=11.8 ms
64 bytes from 8.8.8.8: icmp_seq=3457 ttl=121 time=10.8 ms
64 bytes from 8.8.8.8: icmp_seq=3458 ttl=121 time=61.6 ms
64 bytes from 8.8.8.8: icmp_seq=3459 ttl=121 time=51.0 ms
64 bytes from 8.8.8.8: icmp_seq=3460 ttl=121 time=89.5 ms
64 bytes from 8.8.8.8: icmp_seq=3461 ttl=121 time=58.3 ms
64 bytes from 8.8.8.8: icmp_seq=3462 ttl=121 time=57.8 ms
64 bytes from 8.8.8.8: icmp_seq=3463 ttl=121 time=57.3 ms
64 bytes from 8.8.8.8: icmp_seq=3464 ttl=121 time=70.6 ms
64 bytes from 8.8.8.8: icmp_seq=3465 ttl=121 time=65.4 ms

```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Version: version:  1.15.2

**Additional context**
I do not have another VPN to reproduce with.

**Comment by burtonator** [comment]
Wow.. thanks for reporting this. Has anyone else noticed this problem?

Literally the ONLY thing we're doing that could be causing this is to startup the HTTP daemon which serves the local Polar app...

I mean there could be a bug or something related to express but we're not explicitly doing anything *weird* that would cause this.

the only thing I could think about is maybe your VPN gets confused and thinks Polar is like a proxy server and tries to proxy through it ? (thinking out loud here).

Does this have anything to do with the number of documents?  

You said "almost-empty polar-bookshelf" ... does it work with more documents or are you just asserting that this shouldn't happen because of lower load.

For what it's worth we're TRYING to get rid of the requirement for having a daemon at all but not there yet.

**Comment by bvborisov** [comment]
Hi, thanks for the quick response.

> Does this have anything to do with the number of documents?
Not sure, I tried with only 3 documents, 1 one them is ~ 12Mb, the other 2 less than 1 Mb.

> You said "almost-empty polar-bookshelf" ... does it work with more documents or are you just asserting that this shouldn't happen because of lower load.
Asserting that this shouldn't happen because of lower load.

All of this being said, I now tried to reproduce the same way I described above, but the ping is stable and doesn't increase... The only difference that I can see is that during my previous experiment, there was a thin blue line right below the Polar logo, spanning the whole window, and it was full at around 2/3 of the way. Now when I look at it the line is gone. Is it possible that polar was just taking long to upload the document ? The document in question is the 12 Mb one I described, and I noticed the ping lag right after uploading it.
